src/canvas/Geometry/FaceFinder.tsx

```diff
@@ -21,23 +23,23 @@ export class FaceFinder {
      * face in the inputted segments array. The inner array's order indicates
      * how to go around the face in the negative direction.
      */
-    static FindFaces = (paths: PatternPath[]): number[][] => {
+    static FindFaces = (paths: PatternPath[]): { index: number; isReversed: boolean }[][] => {
         const vertices = FaceFinder._findVertices(paths);
         const edges = FaceFinder._findEdges(paths, vertices);
 
         // Create a map between vertices and an ordered array of all its
         // outgoing edges, ordered by angle of departure from the vertex
         const leavingEdgesMap = FaceFinder._createLeavingEdgesMap(vertices, edges);
 
-        const faces = new Array<Array<number>>();
+        const faces: { index: number; isReversed: boolean }[][] = [];
```

Comment:
Yeah, I really wanted to do that but I never found a name I liked :). Edge is already taken. maybe EdgeDirection? EdgeIndex? EdgeIndexAndDirection? Or just return the whole Edge with all its extraneous information? What do you think?

Reply:
I think it makes more sense to return the whole Edge. Can we move the Edge interface to the types folder?

Reply:
I did that, and while I was there I had a thought. Since FaceFinder is now coupled with PatternPath (since paths hold many segments), and PatternPath now has a bunch of useful methods like getTangentAtStart and getTangentAtEnd, why not simplify the whole thing and have the FaceFinder return directly the PatternPaths needed to form a PatternPiece. I made the changed, tell me what you think.

Reply:
I also updated the FaceFinder.test. To test the new FaceFinder implementation, I needed .equals on PatternPath and on Segments, so I added that too.

Reply:
Cool, it looks good!